src/main/java/com/mixpanel/android/util/HttpService.java

```diff
@@ -27,8 +27,36 @@
  * An HTTP utility class for internal use in the Mixpanel library. Not thread-safe.
  */
 public class HttpService implements RemoteService {
+
+    private static boolean sAdBlockerEnabled;
+
+    @Override
+    public void checkAdBlockerEnabled() {
+        Thread t = new Thread(new Runnable() {
+            public void run() {
+                try {
+                    InetAddress apiMixpanelInet = InetAddress.getByName("api.mixpanel.com");
+                    InetAddress decideMixpanelInet = InetAddress.getByName("decide.mixpanel.com");
+                    sAdBlockerEnabled = apiMixpanelInet.isLoopbackAddress() ||
+                            apiMixpanelInet.isAnyLocalAddress() ||
+                            decideMixpanelInet.isLoopbackAddress() ||
+                            decideMixpanelInet.isAnyLocalAddress();
+                    if (sAdBlockerEnabled ) {
+                        Log.i(LOGTAG, "AdBlocker is enabled. Won't be able to use Mixpanel services.");
+                    }
+                } catch (UnknownHostException e) {
+                }
+            }
+        });
+
+        t.start();
+    }
+
     @Override
     public boolean isOnline(Context context) {
+        if (sAdBlockerEnabled)
```

Comment:
The only scenario where I don't include braces is when doing sanity checks at the beginning of a method. It's easier to read. Alternatively, you can also do one-line sanity checks (e.g. https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/UpdateDisplayState.java#L305). I'll go with the later to make it consistent.